xdsl/dialects/riscv.py

```diff
@@ -624,17 +652,13 @@ def __init__(
         self,
         immediate: int | IntegerAttr | str | LabelAttr,
         *,
-        rd: IntRegisterType | str | None = None,
```

Comment:
allowing str to be passed here was also a mistake, I think, we never used it in practice, and it just makes the code bloated